core/primitives/impl/scale_codec_impl.cpp

```diff
@@ -336,6 +383,42 @@ namespace kagome::primitives {
     return decodeVariant<Invalid, Valid, Unknown>(stream);
   }
 
+  outcome::result<Buffer> ScaleCodecImpl::encodeInherentData(
+      const InherentData &inherentData) const {
+    auto &data = inherentData.getDataCollection();
+
+    // TODO(Harrm) PRE-153 Change to vectors of ref wrappers to avoid copying
+    // vectors
+    std::vector<InherentData::InherentIdentifier> ids;
+    ids.reserve(data.size());
+    std::vector<std::vector<uint8_t>> vals;
+    vals.reserve(data.size());
+
+    for (auto &pair : data) {
+      ids.push_back(pair.first);
+      vals.push_back(std::cref(pair.second.toVector()));
```

Comment:
Are you pushing constref to a vector of values? `std::cref` looks unnecessary here, a copy takes place anyway

Reply:
You are completely right, it was left here after my attempt to make it a vector of references, which failed because of the way SCALE is implemented now. Fixed.